app/assets/javascripts/components/revisions/diff_viewer.jsx

```diff
@@ -156,11 +164,43 @@ const DiffViewer = React.createClass({
     const wikiDiffUrl = this.webDiffUrl();
 
     let diffComment;
-    // Only show the edit summary for a single revision; for cumulative diffs,
-    // the edit summary is for the *parent* of the first revision, so it's
-    // irrelevant and confusing to show that.
+    let revisionDateTime;
+    let firstRevTime;
+    let lastRevTime;
+    let days;
+    let hours;
+    let minutes;
+    let timeSpan = '';
+
+    // Edit summary for a single revision:
+    //  > Edit date and number of characters added
+    // Edit summary for range of revisions:
+    //  > Span of time for edits to article (from first applicable rev to last)
     if (!this.props.first_revision) {
-      diffComment = <p className="diff-comment">{this.state.comment}</p>;
+      revisionDateTime = moment(this.props.revision.date).format('YYYY/MM/DD h:mm a');
+      diffComment = <p className="diff-comment">{this.state.comment}&nbsp;
+                       (Edited on {revisionDateTime};&nbsp;
+                       {this.props.revision.characters} chars changed)</p>;
+    } else {
+      firstRevTime = moment(this.state.firstRevDateTime);
+      lastRevTime = moment(this.state.lastRevDateTime);
+      days = lastRevTime.diff(firstRevTime, 'days');
+      hours = lastRevTime.diff(firstRevTime, 'hours') - days * 24;
+      minutes = lastRevTime.diff(firstRevTime, 'minutes') - (days * 24 + hours) * 60;
+      if (days !== 0) {
+        timeSpan = `${days} day${days > 1 ? 's' : ''}, `;
+      }
+      if (hours !== 0) {
+        timeSpan += `${hours} hour${hours > 1 ? 's' : ''}, `;
+      }
+      if (minutes !== 0) {
+        timeSpan += `${minutes} minute${minutes > 1 ? 's' : ''}`;
+      }
+      if (!timeSpan) {
+        timeSpan = 'less than a minute';
```

Comment:
Yes, that's exactly what I was doing.  There's probably a package for this(?).  In any case, this is how I interpreted this from the story:```Time range for cumulative diff```

I will change to just show the start and end date-times.

Re I18n - do I need to just add something to the .en file?